lindi/LindiH5pyFile/LindiH5pyAttributes.py

```diff
@@ -46,14 +54,14 @@ def __getitem__(self, key):
 
             # Convert special float values to actual floats (NaN, Inf, -Inf)
             # Note that string versions of these values are not supported
-            val = _decode_nan_inf_ninf_in_attr_val(val)
+            val = decode_nan_inf_ninf(val)
 
-            return val
+            return zarr_to_h5_attr(val)
         else:
             raise ValueError(f"Unknown attrs_type: {self._attrs_type}")
 
     def __setitem__(self, key, value):
-        raise KeyError("Cannot set attributes on read-only object")
+        self._write.__setitem__(key, value)
```

Comment:
This is more of a stylistic suggestion than functional. I think it would be more intuitive to have `__init__` be:
```python
    def __init__(self, attrs, attrs_type: Literal["h5py", "zarr"], readonly: bool):
        self._attrs = attrs
        self._attrs_type = attrs_type
        self._readonly = readonly
        if readonly:
            self._writer = LindiH5pyAttributesWrite(self)
```
and this function to be:
```python
    def __setitem__(self, key, value):
        if self._readonly:
            raise KeyError("Cannot set attributes on read-only object")
        self._writer.__setitem__(key, value)
```
(Move the readonly check here and name the object `writer` instead of `write`.)
This mirrors the readonly check in `LindiH5pyDataset.__setitem__` and `LindiH5pyGroup.__setitem__`.